lib/src/core/ether_amount.dart

```diff
@@ -1,49 +1,17 @@
-enum EtherUnit {
-  ///Wei, the smallest and atomic amount of Ether
-  wei,
-
-  ///kwei, 1000 wei
-  kwei,
-
-  ///Mwei, one million wei
-  mwei,
-
-  ///Gwei, one billion wei. Typically a reasonable unit to measure gas prices.
-  gwei,
-
-  ///szabo, 10^12 wei or 1 μEther
-  szabo,
-
-  ///finney, 10^15 wei or 1 mEther
-  finney,
-
-  ether
-}
+part of webthree;
 
 /// Utility class to easily convert amounts of Ether into different units of
 /// quantities.
 class EtherAmount {
-  static final Map<EtherUnit, BigInt> _factors = {
-    EtherUnit.wei: BigInt.one,
-    EtherUnit.kwei: BigInt.from(10).pow(3),
-    EtherUnit.mwei: BigInt.from(10).pow(6),
-    EtherUnit.gwei: BigInt.from(10).pow(9),
-    EtherUnit.szabo: BigInt.from(10).pow(12),
-    EtherUnit.finney: BigInt.from(10).pow(15),
-    EtherUnit.ether: BigInt.from(10).pow(18)
-  };
-
-  final BigInt _value;
-
-  BigInt get getInWei => _value;
-  BigInt get getInEther => getValueInUnitBI(EtherUnit.ether);
-
   const EtherAmount.inWei(this._value);
 
   EtherAmount.zero() : this.inWei(BigInt.zero);
 
   /// Constructs an amount of Ether by a unit and its amount. [amount] can
-  /// either be a base10 string, an int, or a BigInt.
+  /// either be a base10 string, an int or a BigInt.
+  @Deprecated(
```

Comment:
why would we want to deprecate this?

Reply:
To improve visibility of the code with a typed method